compiler/src/dotty/tools/dotc/core/DenotTransformers.scala

```diff
@@ -60,11 +60,14 @@ object DenotTransformers {
   /** A transformer that only transforms SymDenotations */
   trait SymTransformer extends DenotTransformer {
 
+  	/** Tramsform the info of a denotation that is not a Symdenotation */
+    def transformInfo(tp: Type, sym: Symbol)(implicit ctx: Context): Type = tp
+
     def transformSym(sym: SymDenotation)(implicit ctx: Context): SymDenotation
 
     def transform(ref: SingleDenotation)(implicit ctx: Context): SingleDenotation = ref match {
       case ref: SymDenotation => transformSym(ref)
-      case _ => ref
+      case _ => ref.derivedSingleDenotation(ref.symbol, transformInfo(ref.info, ref.symbol))
```

Comment:
Isn't `derivedSingleDenotation` going to force the info of the ref?

Reply:
Ah in fact ref.info on that line does that more directly :). That means we're suddenly forcing all infos of non-sym SingleDenotations which doesn't sound ideal.

Reply:
Only for opaque helper types. So it's not out of the ordinary,

Reply:
You are right. Let's make it a DenotTransformer instead.